### PR TITLE
[main][fix] Change withdraw protocol reserve restriction

### PR DESCRIPTION
**Pull request by m100alpaca**
## Description
- Change withdrawProtocolReserve restriction from only owner to whitelist

## Why?
- Allow whitelisted callers to withdraw protocol reserve instead of owner only

### CODE REVIEW

File: solidity/contracts/money-market/facets/AdminFacet.sol [hunk 1]
```diff
@@ -30,6 +30,8 @@ contract AdminFacet is IAdminFacet {
   using SafeCast for uint256;
   using LibDoublyLinkedList for LibDoublyLinkedList.List;
 
+  mapping(address => bool) public whitelistedCallers;
```
Comment (alpanaca) [inline solidity/contracts/money-market/facets/AdminFacet.sol:33]: Move this to LibMoneyMarket01 and change name to operatorsOk. Make sure it's at the end of struct ordering

File: solidity/contracts/money-market/libraries/LibMoneyMarket01.sol [hunk 2]
```diff
@@ -132,6 +132,7 @@ library LibMoneyMarket01 {
     mapping(address => bool) liquidatorsOk; // allowed to initiate liquidation process
     mapping(address => bool) accountManagersOk; // allowed to manipulate account/subaccount on behalf of end users
     mapping(address => bool) riskManagersOk; // allowed to set risk control parameters
+    mapping(address => bool) operatorsOk; // allowed to withdraw the protocol reserve
```
Comment (alpanaca) [inline solidity/contracts/money-market/libraries/LibMoneyMarket01.sol:135]: move this to end of struct

File: solidity/contracts/money-market/facets/AdminFacet.sol [hunk 3]
```diff
@@ -545,12 +546,30 @@ contract AdminFacet is IAdminFacet {
     emit LogSetEmergencyPaused(msg.sender, _isPaused);
   }
 
+  /// @notice Set whitelisted callers
+  /// @param _callers The addresses of the callers that are going to be whitelisted.
+  /// @param _allow Whether to allow or disallow callers.
+  function setWhitelistedCallers(address[] calldata _callers, bool _allow) external onlyOwner {
```
Comment (alpanaca) [inline solidity/contracts/money-market/facets/AdminFacet.sol:552]:
```suggestion
  function setOperatorsOk(address[] calldata _callers, bool _allow) external onlyOwner {
```

File: solidity/contracts/money-market/facets/AdminFacet.sol [hunk 4]
```diff
@@ -545,12 +546,30 @@ contract AdminFacet is IAdminFacet {
     emit LogSetEmergencyPaused(msg.sender, _isPaused);
   }
 
+  /// @notice Set whitelisted callers
+  /// @param _callers The addresses of the callers that are going to be whitelisted.
+  /// @param _allow Whether to allow or disallow callers.
+  function setOperatorsOk(address[] calldata _callers, bool _allow) external onlyOwner {
```
Comment (wow-alpaca) [inline solidity/contracts/money-market/facets/AdminFacet.sol:552]: if we change from `whitelistedCallers` to `operatorsOk` should we change the name of param (also natspec)?

**Review by alpanaca — DISMISSED** [review]

**Review by wow-alpaca — APPROVED** [review]

**Review by wow-alpaca — DISMISSED** [review]

**Review by alpanaca — DISMISSED** [review]

File: solidity/contracts/money-market/libraries/LibMoneyMarket01.sol [hunk 5]
```diff
@@ -152,6 +152,8 @@ library LibMoneyMarket01 {
     uint16 liquidationFeeBps; // fee that is charged during liquidation by protocol, goes to liquidationTreasury
     uint16 flashloanFeeBps; // fee that is charged when providing the flashloan
     uint16 lenderFlashloanBps; // portion of flashloan fee that will go to lenders
+    // Additional state
+    mapping(address => bool) operatorsOk; // allowed to withdraw the protocol reserve
```
Comment (jr-alpaca) [inline solidity/contracts/money-market/libraries/LibMoneyMarket01.sol:156]: move it under flashloanTreasury

**Review by wow-alpaca — APPROVED** [review]

**Review by jr-alpaca — APPROVED** [review]

**Pull request merged**